src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -3891,5 +3953,950 @@ void LocalFunction()
             CreateSubmission(source, previous: previous)
                 .VerifyEmitDiagnostics();
         }
+
+        [Fact]
+        public void ShadowNames_01()
+        {
+            var source =
+@"#pragma warning disable 0219
+#pragma warning disable 8321
+class Program
+{
+    static void M()
+    {
+        void F1(object x) { string x = null; } // local
+        void F2(object x, string y, int x) { } // parameter
+        void F3(object x) { void x() { } } // method
+        void F4<x, y>(object x) { void y() { } } // type parameter
+        void F5(object M, string Program) { }
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular7_3);
+            verifyDiagnostics();
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            verifyDiagnostics();
+
+            comp = CreateCompilation(source);
```

Comment:
How is this different than the `CreateCompilation` call above?

Reply:
It's not today but it will be in the future when the latest version number increases. I wanted at least one test that explicitly tested C#8 rather than (or in addition to) latest.

---
In reply to: [245372913](https://github.com/dotnet/roslyn/pull/32067#discussion_r245372913) [](ancestors = 245372913)